### PR TITLE
feat: [CDS-72503]: Create a new step for Revert Merged PR

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by ishantmehtaharness** [comment]
retrigger titlecheck

**Comment by ishantmehtaharness** [comment]
Branch coverage is failing, I will add more coverage as part of this ticket
https://harness.atlassian.net/browse/CDS-73688
Screenshot 2023-07-05 at 8 18 28 PM

File: src/modules/75-cd/components/GitOpsRevertPR/__tests__/RevertPRStep.test.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+describe('Test Revert PR Step', () => {
+  beforeEach(() => {
+    factory.registerStep(new GitOpsRevertPR())
+  })
+  test('should render edit view as new step', () => {
+    const { container } = render(
+      <TestStepWidget initialValues={{}} type={StepType.RevertPR} stepViewType={StepViewType.Edit} />
+    )
+    expect(container).toMatchSnapshot()
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/GitOpsRevertPR/__tests__/RevertPRStep.test.tsx:28]: Let's avoid snapshot testing as much as possible. In fact, we will be better off not having it.

File: src/modules/75-cd/components/GitOpsRevertPR/GitOpsRevertPR.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    viewType
+  }: ValidateInputSetProps<RevertPRStepData>): FormikErrors<RevertPRStepData> {
+    return validateGitOpsExecutionStepForm({ data, template, getString, viewType })
+  }
+
+  processFormData(values: RevertPRStepData): RevertPRStepData {
+    return values
+  }
+
+  protected type = StepType.RevertPR
```
[the remark this answers is not in the document]
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/GitOpsRevertPR/GitOpsRevertPR.tsx:102]: Yes, we can do that.
in next PR

File: src/modules/75-cd/components/GitOpsRevertPR/GitOpsRevertPRInputStep.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import { isEmpty } from 'lodash-es'
+import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
+import { useStrings } from 'framework/strings'
+import { TimeoutFieldInputSetView } from '@pipeline/components/InputSetView/TimeoutFieldInputSetView/TimeoutFieldInputSetView'
+import { isExecutionTimeFieldDisabled } from '@pipeline/utils/runPipelineUtils'
+import { useVariablesExpression } from '@pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression'
+import type { GitOpsRevertPRProps } from './helper'
+import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
+
+function GitOpsRevertPRInputStep({
```
[the remark this answers is not in the document]
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/GitOpsRevertPR/GitOpsRevertPRInputStep.tsx:19]: why Chetan?

File: src/modules/75-cd/components/GitOpsRevertPR/__tests__/RevertPRStep.test.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/75-cd/components/GitOpsRevertPR/__tests__/RevertPRStep.test.tsx:28]: I find it useful to do atleast few testing with snapshots
I have seen it being beneficial

**Review by sahithikolichala — APPROVED** [review]

**Review by cjlee01 — APPROVED** [review]

File: src/modules/75-cd/components/GitOpsRevertPR/GitOpsRevertPRInputStep.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/GitOpsRevertPR/GitOpsRevertPRInputStep.tsx:19]: If you use word `Step`, it looks like that it is parent level Step component. So, use `Step` word only for parent component. And include words like `Edit` and `InputSet` for edit and input set components respectively.
So, your component names could have been like - `GitOpsRevertPRStep`, `GitOpsRevertPREdit`, `GitOpsRevertPRInputSet`

File: src/modules/75-cd/components/GitOpsRevertPR/__tests__/RevertPRStep.test.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/GitOpsRevertPR/__tests__/RevertPRStep.test.tsx:28]: It is only beneficial when we do not even open browser and see how our UI is looking. Other than CSS, it does not give any guarantee. Your functionality can easily be broken and your snapshots will not catch anything. If you are aiming to cover lines and statements faster, that can also be done using simple assertions.

**Review by chetanvekariya-harness — APPROVED** [review]
Approving as reported comments are more around naming conventions and can be addressed in the next PR.

**Pull request merged**